gensim/matutils.py

```diff
@@ -382,6 +383,110 @@ def cossim(vec1, vec2):
     return result
 
 
+def isbow(vec):
+    """
+    Checks if vector passed is in bag of words representation or not.
+    """
+    if scipy.sparse.issparse(vec):
+        vec = vec.todense().tolist()
+    try:
+        id_, val_ = vec[0]
+        id_, val_ = int(id_), float(val_)
+    except Exception:
+        return False
+    return True
+
+
+def kullback_leibler(vec1, vec2, num_features=None):
+    """
+    A similarity metric between two probability distributions.
+    Returns a similarity in range <0,1> where values closer to 0 mean a higher similarity.
+    Uses the scipy.stats.entropy method to identify kullback_leibler convergence value.
+    If the distribution draws from a certain number of docs, that value must be passed.
+    """
+    if scipy.sparse.issparse(vec1):
+        vec1 = numpy.asarray(vec1.todense())
+    if scipy.sparse.issparse(vec2):
+        vec2 = numpy.asarray(vec2.todense())
+    if isinstance(vec1, numpy.ndarray):
+        vec1 = vec1.tolist()
+    if isinstance(vec2, numpy.ndarray):
+        vec2 = vec2.tolist()
+    if isbow(vec1) and isbow(vec2):
```

Comment:
I think a good course of action is: make the simplest possible version, make sure it's correct (unittests). For example, this could be converting the inputs (no matter its format) into two numpy arrays and then performing the calculation on that.

Once this version is debugged and working, profile it for different types of input, and perhaps optimize for different code-paths (such as when both inputs are scipy.sparse, or both are bow, etc).